### ISSUE TITLE
make_fx loses node.stack_trace / turn on AOTAutograd by default for all backends

### ISSUE THREAD

**Issue by silvasean** [pytorch/pytorch#90276]
### 🐛 Describe the bug

It appears that make_fx loses the node.stack_trace on the relevant nodes. This makes it difficult to integrate into a production compiler since it makes debugging of the resulting model much more difficult. It would be nice if make_fx preserved the node.stack_trace.

```
import torch
import torch._dynamo as dynamo
import functorch

from typing import List

def my_backend(gm: torch.fx.GraphModule,
               example_inputs: List[torch.Tensor]):
    gm.print_readable()
    make_fx_gm = functorch.make_fx(gm)(*example_inputs)
    make_fx_gm.print_readable()
    return make_fx_gm

@dynamo.optimize(my_backend)
def f(x):
    return x * x

f(torch.rand(3))
```

Output:

```
# The graph captured by TorchDynamo
class GraphModule(torch.nn.Module):
    def forward(self, x : torch.Tensor):
        # File: /tmp/repro2.py:16, code: return x * x
        mul = x * x;  x = None
        return (mul,)

# The result of make_fx
class <lambda>(torch.nn.Module):
    def forward(self, arg0_1: f32[3]):
        # No stacktrace found for following nodes <------------ missing node.stack_trace
        mul: f32[3] = torch.ops.aten.mul.Tensor(arg0_1, arg0_1);  arg0_1 = None
        return (mul,)
        
```

### Versions

Collecting environment information...
PyTorch version: 1.14.0.dev20221201+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Debian GNU/Linux rodete (x86_64)
GCC version: (Debian 12.2.0-3) 12.2.0
Clang version: 14.0.6-2
CMake version: version 3.25.0
Libc version: glibc-2.35

Python version: 3.10.8 (main, Nov  3 2022, 15:17:13) [GCC 12.2.0] (64-bit runtime)
Python platform: Linux-5.19.11-1rodete1-amd64-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.24.0rc1
[pip3] torch==1.14.0.dev20221201+cpu
[pip3] torchvision==0.15.0.dev20221201+cpu
[conda] Could not collect

cc @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519 @gchanan @soumith @ngimel

**Comment by SherlockNoMad** [comment]
We actually didn't expect backend to directly call make_fx. 
I am wondering why do you need to call make_fx here? Are you expecting to get a FX-Aten graph somehow?

**Comment by ezyang** [comment]
Oh no, we need to fix this API.

For now, what you actually want to do is wrap your backend in AOTAutograd. What this will do is not only normalize the Torch calls in the graph to ATen, it will also functionalize your graph (no more need for mutation handling) and turn on training support.

It's pretty easy to do: just replace `my_backend` with `aot_autograd(fw_compiler=my_backend)` (get `aot_autograd` from `torch._dynamo.optimizations.training`). There are some other knobs which you can look at in the file in question.

What we actually need to do is apply aot autograd BY DEFAULT for backend compilers, and force you to opt out of aot autograd if you don't want it. We'll do that soon.

**Comment by silvasean** [comment]
Sorry for the delay getting back on this.

Thanks for the feedback @ezyang and @SherlockNoMad. Yes, we are using make_fx to convert to aten ops. I migrated us to use aot_autograd as Ed suggested.

Ed mentioned that we could get functionalization as well (which we definitely want too!), but I didn't see that happening with aot_autograd. I put more details in this issue: https://github.com/pytorch/pytorch/issues/90759

**Comment by kshitij12345** [comment]
Given that, `aot_autograd` (preferred approach for getting aten-fx graph) tracks the stack_trace (see below), can this issue be closed?

```python
import torch
import torch._dynamo as dynamo
from torch._dynamo.backends.common import aot_autograd

from typing import List

def my_backend(gm: torch.fx.GraphModule,
                    example_inputs: List[torch.Tensor]):
    gm.print_readable()
    return gm

my_backend = aot_autograd(fw_compiler=my_backend) 

@dynamo.optimize(backend=my_backend)
def f(x):
    return x * x

print(f(torch.rand(3)))

```

Output
```
class <lambda>(torch.nn.Module):
    def forward(self, arg0_1: f32[3]):
        # File: test/test_scratch.py:321, code: return x.norm()
        linalg_vector_norm: f32[] = torch.ops.aten.linalg_vector_norm.default(arg0_1);  arg0_1 = None
        return (linalg_vector_norm,)
        
tensor(1.4554)
```

**Comment by ydwu4** [comment]
I'll close the issue for now but feel free to re-open when necessary.
